javascript/javascript1/week4/homework/Voice_Assistant.js

```diff
@@ -0,0 +1,154 @@
+    
+    if (command === "Add fishing to my todo") {
+        toDoArray.push("fishing");
+        return "Fishing added to your todo.";
+    }
+
+    if (command === "Add singing in the shower to my todo") {
+        toDoArray.push("singing in the shower");
+        return "Singing in the shower added to your todo.";
+    }
+
+    if (command === "Remove fishing from my todo") {
+        for (let i = 0; i < toDoArray.length; i++) {
+            if (toDoArray[i] === "fishing") { 
+                toDoArray.splice(i, 1);
+                return "Removed fishing from your todo."; 
+            }
+        }
+    }
+
+    if (command === "What is on my todo?") {
+        return toDoArray.join(", ");
+    }
+
+    if (command === "What day is it today?") {
+        let today = new Date();
+        let day = today.getDate();
+        let months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+        let month = months[today.getMonth()];
+        let year = today.getFullYear(); 
+
+        return `${day}. of ${month}. of ${year}`;
+    }
+
+
+    if (command.startsWith("What is ")) {
+        if (command.includes("+")) {
+            let numbers = command.replace("What is ", "").split(" + "); 
+            let sum = parseInt(numbers[0]) + parseInt(numbers[1]);
+            return sum;
+
+        } else if (command.includes("-")) {
+            let numbers = command.replace("What is ", "").split(" - "); 
+            let difference = parseInt(numbers[0]) - parseInt(numbers[1]);
+            return difference;
+
+        } else if (command.includes("*")) {
+            let numbers = command.replace("What is ", "").split(" * "); 
+            let product = parseInt(numbers[0]) * parseInt(numbers[1]);
+            return product;
+
+        } else if (command.includes("/")) {
+            let numbers = command.replace("What is ", "").split(" / "); 
+            let quotient = parseInt(numbers[0]) / parseInt(numbers[1]);
+            return quotient;
+        }
+    }
+
+    if (command.startsWith("Set a timer for ")) {
+        let minutesText = command.replace("Set a timer for ", ""); 
+        let minutes = parseInt(minutesText); 
+    
+        if (isNaN(minutes) || minutes <= 0) { 
+            return "Please provide a valid number of minutes.";
+        }
+    
+        setTimeout(function () {
+            console.log("Timer done!"); 
+        }, minutes * millisecondsInAMinute); 
+    
+        return `Timer set for ${minutes} minutes.`;
+    }
+
+    return "I don't understand that command.";
+}
+
+
+
+console.log(getReply("Hello my name is Benjamin")); 
+console.log(getReply("What is my name?")); 
+console.log(getReply("Hello my name is Benjamin")); 
+console.log(getReply("Hello my name is Anita")); 
+console.log(getReply("What is my name?"));
+console.log(getReply("Hello my name is Anita"));
+console.log(getReply("Hello my name is")); // Not sure why this does not work
```

Comment:
```js
"Hello my name is".startsWith("Hello my name is ") // false
```
My suggestion is to remove the space from the command. Strings are super easy to get wrong (the devil is in the details :smiling_imp:)